src/controllers/utils/advanced-search-validation.ts

```diff
@@ -74,43 +85,133 @@ export const advancedSearchValidationRules =
                 }
                 return true;
             }),
-        check(DISSOLVED_FROM_FIELD)
+        check(DISSOLVED_FROM_DAY_FIELD)
             .custom((date, { req }) => {
-                if (isStringNotNullOrEmpty(date)) {
-                    if (!isDateFormattedProperly(date)) {
-                        throw Error(DISSOLVED_INVALID_DATE_ERROR_MESSAGE);
+                const dissolvedFromDay = req.query?.dissolvedFromDay as string;
+                const dissolvedFromMonth = req.query?.dissolvedFromMonth as string;
+                const dissolvedFromYear = req.query?.dissolvedFromYear as string;
+
+                // check that one part of the date is not empty and check if day is missing
+                if (isStringNotNullOrEmpty(dissolvedFromYear) || isStringNotNullOrEmpty(dissolvedFromMonth)) {
+                    if (!isStringNotNullOrEmpty(dissolvedFromDay)) {
+                        throw Error(DISSOLVED_FROM_DATE_MUST_INCLUDE_DAY);
```

Comment:
Perhaps a named function might be used to replace these duplicate code blocks, make the code briefer and make the purpose and function clearer.

Reply:
separated out into separate functions, thanks